apps/meteor/client/views/room/MessageList/components/MessageIndicators.tsx

```diff
@@ -1,55 +1,47 @@
-import { IMessage, isEditedMessage } from '@rocket.chat/core-typings';
-import { MessageStatusIndicator, MessageStatusIndicatorItem } from '@rocket.chat/fuselage';
+import { IMessage, isEditedMessage, ITranslatedMessage } from '@rocket.chat/core-typings';
+import { Box, MessageStatusIndicator, MessageStatusIndicatorItem } from '@rocket.chat/fuselage';
 import { useUserId, useTranslation } from '@rocket.chat/ui-contexts';
 import React, { FC } from 'react';
 
 import { isE2EEMessage } from '../../../../../lib/isE2EEMessage';
-import { useMessageDateFormatter, useShowStarred, useShowTranslated, useShowFollowing } from '../contexts/MessageListContext';
-
-// edited() {
-// 	const { msg } = this;
-// 	return msg.editedAt && !MessageTypes.isSystemMessage(msg);
-// },
-// editTime() {
-// 	const { msg } = this;
-// 	return msg.editedAt ? formatDateAndTime(msg.editedAt) : '';
-// },
-// editedBy() {
-// 	const { msg } = this;
-// 	if (!msg.editedAt) {
-// 		return '';
-// 	}
-// 	// try to return the username of the editor,
-// 	// otherwise a special "?" character that will be
-// 	// rendered as a special avatar
-// 	return (msg.editedBy && msg.editedBy.username) || '?';
-// },
+import {
+	useMessageDateFormatter,
+	useShowStarred,
+	useShowTranslated,
+	useShowFollowing,
+	useTranslateProvider,
+} from '../contexts/MessageListContext';
 
 export const MessageIndicators: FC<{
-	message: IMessage;
+	message: IMessage & Partial<ITranslatedMessage>;
 }> = ({ message }) => {
 	const t = useTranslation();
-	const translated = useShowTranslated({ message }); // TODO: useMessageTranslated
-	const starred = useShowStarred({ message }); // TODO: useMessageStarred
-	const following = useShowFollowing({ message }); // TODO: useMessageFollowing
+	const translated = useShowTranslated({ message });
+	const translateProvider = useTranslateProvider({ message });
+	const starred = useShowStarred({ message });
+	const following = useShowFollowing({ message });
 
 	const isEncryptedMessage = isE2EEMessage(message);
 	const uid = useUserId();
 
-	const formatter = useMessageDateFormatter(); // TODO: useMessageDateFormatter
+	const formatter = useMessageDateFormatter();
 
 	return (
 		<MessageStatusIndicator>
-			{translated && <MessageStatusIndicatorItem name='pin' data-title={t('Message_has_been_pinned')} />}
+			{translated && (
+				<Box is='span' color='info' fontSize='micro' withTruncatedText>
+					<MessageStatusIndicatorItem name='language' title={t('Translated')} /> {translateProvider}
+				</Box>
```

Comment:
This component will be added on Fuselage on this PR: https://github.com/RocketChat/fuselage/pull/757